web/src/redux/slices/TestDefinition.slice.ts

```diff
@@ -56,6 +56,11 @@ const testDefinitionSlice = createSlice({
         return def;
       });
     },
+    revertDefinition(state, {payload: {index}}: PayloadAction<{index: number}>) {
+      state.definitionList = state.definitionList
+        .map((d, idx) => (idx === index ? state.initialDefinitionList[index] || undefined : d))
```

Comment:
I think you could achieve the same behavior using reduce, something like:

```
const {selector} = payload;
state.definitionList.reduce((list, definition) => {
  const foundDef  = state.initialDefinitionList.find((def) => def.selector === selector);
  
  return foundDef ? list.concat([foundDef]) : list;
}, []);
```

Reply:
I have tried the selector approach, but the selector can change.
So what you suggested does not work